src/kbucket.rs

```diff
@@ -1,219 +1,87 @@
-#![allow(unused)]
-
-use crate::helper::{Identifier, Node, U256};
+use crate::helper::{Identifier, U256};
 use std::collections::HashMap;
 use std::net::Ipv4Addr;
-use uint::*;
 
 const BUCKET_SIZE: usize = 20;
 const MAX_BUCKETS: usize = 256;
 
-type Bucket = [Option<Node>; BUCKET_SIZE];
+// Should I create a HashMap with a generic K, V or have my hashmap take in an id and TR?
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct Bucket {
+    pub map: HashMap<Identifier, TableRecord>,
```

Comment:
the way this works right now is that you have a map from `Identifier` to a struct that contains the same `Identifier`

I'd avoid duplicating the information unless you really need to do so

a `Bucket` really is just a map from node id's to the corresponding (ip, port) right?

Reply:
yep yep